core/src/states/context.rs

```diff
@@ -86,6 +87,12 @@ pub enum Duty {
     },
 }
 
+/// Result of handling a duty
+#[derive(Debug, Clone, Default)]
+pub struct DutyResult {
```

Comment:
I feel like this is better as an enum. Though, I'm not sure if we're gonna add anything else to it later on or if we'll need to combine multiple enum items.